features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemLocationView.kt

```diff
@@ -28,26 +30,36 @@ import io.element.android.features.messages.impl.timeline.model.event.TimelineIt
 import io.element.android.features.messages.impl.timeline.model.event.TimelineItemLocationContentProvider
 import io.element.android.libraries.designsystem.preview.ElementPreviewDark
 import io.element.android.libraries.designsystem.preview.ElementPreviewLight
+import io.element.android.libraries.designsystem.theme.components.Text
 
 @Composable
 fun TimelineItemLocationView(
     content: TimelineItemLocationContent,
     modifier: Modifier = Modifier,
 ) {
-    StaticMapView(
-        modifier = modifier
-            .fillMaxWidth()
-            .heightIn(max = 188.dp),
-        lat = content.location.lat,
-        lon = content.location.lon,
-        zoom = 15.0,
-        contentDescription = content.body
-    )
+    Column(modifier = modifier.fillMaxWidth()) {
+        content.description?.let {
+            Text(
+                text = it,
+                modifier = Modifier.padding(start = 12.dp, end = 12.dp, top = 8.dp, bottom = 8.dp),
+            )
+        }
+
+        StaticMapView(
+            modifier = Modifier
+                .fillMaxWidth()
+                .heightIn(max = 188.dp),
+            lat = content.location.lat,
+            lon = content.location.lon,
+            zoom = 15.0,
+            contentDescription = content.body
+        )
+    }
 }
 
 @Preview
 @Composable
-internal fun TimelineItemLocationViewLightPreview(@PreviewParameter(TimelineItemLocationContentProvider::class) content: TimelineItemLocationContent) =
+internal fun TimelineItemLocationViewPreview(@PreviewParameter(TimelineItemLocationContentProvider::class) content: TimelineItemLocationContent) =
```

Comment:
Sigh sob we'll have to keep the `Light` at the moment...

Reply:
Ah yeah, missed the name when I was rebasing it :sweat_smile: